Junk4Dollars-app/API/AuctionsApiService.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+public class AuctionsApiService {
+    let client: APICallable
```

Comment:
You're using 3 different names for the same concept: `client`, `callable` and `query`, which makes it a little bit confusing.
What do you think about using `client` (with slight variations) for everything?